src/main/java/org/broad/igv/sam/AlignmentTrack.java

```diff
@@ -885,12 +892,16 @@ public boolean handleDataClick(TrackClickEvent te) {
         MouseEvent e = te.getMouseEvent();
         if (Globals.IS_MAC && e.isMetaDown() || (!Globals.IS_MAC && e.isControlDown())) {
             // Selection
-            final ReferenceFrame frame = te.getFrame();
-            if (frame != null) {
-                selectAlignment(e, frame);
-                IGV.getInstance().repaint(this);
-                return true;
+            Alignment alignment = this.getAlignmentAt(te);
+            if (alignment != null) {
+                if (selectedReadNames.containsKey(alignment.getReadName())) {
+                    selectedReadNames.remove(alignment.getReadName());
+                } else {
+                    setSelectedAlignment(alignment);
+                }
+                IGV.getInstance().repaint(this); //todo check if doing this conditionally here is ok
```

Comment:
The logic here was inlined from selectAlignment(e, frame).  The repaint logic changed slightly, previously it would repaint when no alignment was present at the given location but now it only does so if an alignment was selected.  I assume that's ok but i figured I would ask to be sure I didn't miss something.

Reply:
That looks o.k.